assembl/static2/js/app/pages/survey.jsx

```diff
@@ -50,52 +93,51 @@ class Survey extends React.Component {
   }
 
   updateContentLocaleMappingFromProps(props) {
-    const { data, defaultContentLocaleMapping, updateContentLocaleMapping } = props;
-    if (!data.loading) {
-      const contentLocaleMappingData = {};
-      const questions = data.thematic.questions;
-      questions.forEach((question) => {
-        question.posts.edges.forEach((edge) => {
-          const post = edge.node;
-          const { id, originalLocale } = post;
-          const contentLocale = defaultContentLocaleMapping.get(originalLocale, originalLocale);
-          contentLocaleMappingData[id] = {
-            contentLocale: contentLocale,
-            originalLocale: originalLocale
-          };
-        });
+    const { defaultContentLocaleMapping, updateContentLocaleMapping, questions } = props;
+    const contentLocaleMappingData = {};
+    questions.forEach((question) => {
+      question.posts.edges.forEach((edge) => {
+        const post = edge.node;
+        const { id, originalLocale } = post;
+        const contentLocale = defaultContentLocaleMapping.get(originalLocale, originalLocale);
+        contentLocaleMappingData[id] = {
+          contentLocale: contentLocale,
+          originalLocale: originalLocale
+        };
       });
+    });
 
-      updateContentLocaleMapping(contentLocaleMappingData);
-    }
+    updateContentLocaleMapping(contentLocaleMappingData);
   }
 
-  getIfProposals(questions) {
-    this.questions = questions;
-    if (!this.questions) return false;
+  getIfProposals = (questions) => {
+    if (!questions) return false;
     let isProposals = false;
-    this.questions.forEach((question) => {
+    questions.forEach((question) => {
       if (question.posts.edges.length > 0) isProposals = true;
     });
     return isProposals;
-  }
-  showMoreProposals() {
+  };
+
+  showMoreProposals = () => {
     this.setState({
       moreProposals: true
     });
-  }
-  scrollToQuestion(isScroll, questionIndex) {
+  };
+
+  scrollToQuestion = (isScroll, questionIndex) => {
     this.setState({
       isScroll: isScroll,
       questionIndex: questionIndex
     });
-  }
+  };
+
   render() {
-    if (this.props.data.error) {
+    if (this.props.hasError) {
```

Comment:
hasErrors you meant?

Reply:
nice catch

Reply:
done. I don't understand how flow has not seen it !?